app/javascript/graphs/segment_graphs.js

```diff
@@ -0,0 +1,109 @@
+import Highcharts from 'highcharts'
+import Exporting from 'highcharts/modules/exporting'
+Exporting(Highcharts)
+
+const buildSegmentGraphs = function(run) {
+  if (document.getElementById('segment-graphs') === null) {
+    return
+  }
+
+  const url = new URL(window.location.href)
+  const durationString = `${url.searchParams.get('timing') || 'real'}time_duration_ms`
+  const shortestString = `${url.searchParams.get('timing') || 'real'}time_shortest_duration_ms`
+
+  const pbData = []
+  const meanData = []
+  let resetCounter = run.attempts
+  const resetData = []
+
+  run.segments.forEach((segment, i) => {
+    let time = (segment[durationString] - segment[shortestString]) / 1000
+    if (time < 0) {
+      time = 0
+    }
+    pbData.push([segment.name, time])
+
+    time = 0
+    let counter = 0
+    if (segment.histories.length !== 0) {
+      segment.histories.forEach((attempt) => {
+        if (attempt[durationString] === 0) {
+          return
+        } else if (run.segments[i - 1] !== undefined) {
+          const prevSegAttempt = run.segments[i - 1].histories.find((prevAttempt) => {
+            return prevAttempt.attempt_number === attempt.attempt_number
+          })
+          if (prevSegAttempt !== undefined && prevSegAttempt[durationString] === 0) {
+            return
+          }
+        }
+
+        time += attempt[durationString]
+        counter++
+      })
+
+      time = ((time / counter) - segment[shortestString]) / 1000
+    }
+    meanData.push([segment.name, time])
+
+    resetData.push([segment.name, (1 - (segment.histories.length / resetCounter)) * 100])
+    resetCounter = segment.histories.length
+  })
+
+  Highcharts.chart('segment-graphs', {
+    exporting: {
+      chartOptions: {
+        plotOptions: {
+          series: {
+            dataLabels: {enabled: true}
+          }
+        }
+      },
+      fallbackToExportServer: false
+    },
+    chart: {
+      type: 'column',
+      zoomType: 'x'
+    },
+    title: {text: 'Individual Segment Stats'},
+    subtitle: {text: '(PB - Gold) | (Average - Gold) | (1 - (Finished Attempts / Total Attempts))'},
```

Comment:
Yeah I agree this is awkward, and I think always has been. We should start leaning away from mathematical descriptions and more towards human-friendly stuff. Even for me it's hard to internalize what these bars really *mean* by looking at a simple equation like `PB - Gold`, and now with three this is overload.

The subtitle should be an ELI5 like "Tall is bad". Then we can hide some more words in mouseover tooltips or something like "Difference between PB and best", "Difference between average and best".

In the future maybe we can get more creative and find ways for graphs to speak for themselves. It'd be interesting if we eventually adapted the [fixed placement columns][1] chart to show how averages are approaching bests or something.

[1]: https://www.highcharts.com/demo/column-placement